logOut wirft Exception in FunctionChooser

beim Ausloggen wirft der FunctionChooser eine Exception, sollte man Analyse vorher noch nicht aufgerufen haben.

Stack Trace:
```stacktrace
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at javafx.fxml/javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1857)
        at javafx.fxml/javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(FXMLLoader.java:1724)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
        at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
        at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
        at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
        at javafx.graphics/javafx.scene.Node.fireEvent(Node.java:8792)
        at javafx.controls/javafx.scene.control.Button.fire(Button.java:203)
        at javafx.controls/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:208)
        at javafx.controls/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
        at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
        at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
        at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
        at javafx.base/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.base/javafx.event.Event.fireEvent(Event.java:198)
        at javafx.graphics/javafx.scene.Scene$MouseHandler.process(Scene.java:3897)
        at javafx.graphics/javafx.scene.Scene.processMouseEvent(Scene.java:1878)
        at javafx.graphics/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2623)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:411)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:301)
        at java.base/java.security.AccessController.doPrivileged(Unknown Source)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:450)
        at javafx.graphics/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
        at javafx.graphics/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:449)
        at javafx.graphics/com.sun.glass.ui.View.handleMouseEvent(View.java:557)
        at javafx.graphics/com.sun.glass.ui.View.notifyMouse(View.java:943)
        at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at com.sun.javafx.reflect.Trampoline.invoke(MethodUtil.java:77)
        at jdk.internal.reflect.GeneratedMethodAccessor2.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at javafx.base/com.sun.javafx.reflect.MethodUtil.invoke(MethodUtil.java:275)
        at javafx.fxml/com.sun.javafx.fxml.MethodHelper.invoke(MethodHelper.java:84)
        at javafx.fxml/javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1854)
        ... 46 more
Caused by: java.lang.NullPointerException: Cannot read field "analyseButton" because "github.weichware10.analyse.gui.analyse.Analyzer.controller" is null
        at github.weichware10.analyse.gui.analyse.Analyzer.reset(Analyzer.java:327)
        at github.weichware10.analyse.gui.general.FunctionChooserController.logOut(FunctionChooserController.java:25)
        ... 57 more
```

--- a/src/main/java/github/weichware10/analyse/gui/analyse/Analyzer.java
+++ b/src/main/java/github/weichware10/analyse/gui/analyse/Analyzer.java
@@ -324,23 +324,27 @@ public class Analyzer extends AbsScene {
      * Setzt Analyse zurück.
      */
     public static void reset() {
-        controller.analyseButton.setDisable(true);
-        controller.analyseTypMenuButton.setDisable(true);
-        controller.configButton.setDisable(true);
-        controller.exportButton.setDisable(true);
-        controller.exportRawButton.setDisable(true);
-        controller.selectCompTrialButton.setVisible(false);
-        controller.exportButton.setDisable(true);
-        controller.exportRawButton.setDisable(true);
-        controller.errorLabel.setVisible(false);
-        controller.statusLabel.setVisible(false);
-        controller.analysePane.getChildren().clear();
-        Analyzer.analyseType = null;
-        Analyzer.trial = null;
-        Analyzer.trialComp = null;
-        Analyzer.hmConfig = new HeatmapConfig();
-        Analyzer.diaConfig = new DiagramConfig();
-        Analyzer.heatmapImage = null;
-        Analyzer.verlaufLineChart = null;
+        if (controller != null) {
+            Logger.info("analyzer:content Resetting");
+
+            controller.analyseButton.setDisable(true);
+            controller.analyseTypMenuButton.setDisable(true);
+            controller.configButton.setDisable(true);
+            controller.exportButton.setDisable(true);
+            controller.exportRawButton.setDisable(true);
+            controller.selectCompTrialButton.setVisible(false);
+            controller.exportButton.setDisable(true);
+            controller.exportRawButton.setDisable(true);
+            controller.errorLabel.setVisible(false);
+            controller.statusLabel.setVisible(false);
+            controller.analysePane.getChildren().clear();
+            Analyzer.analyseType = null;
+            Analyzer.trial = null;
+            Analyzer.trialComp = null;
+            Analyzer.hmConfig = new HeatmapConfig();
+            Analyzer.diaConfig = new DiagramConfig();
+            Analyzer.heatmapImage = null;
+            Analyzer.verlaufLineChart = null;
+        }
     }
 }

--- a/src/main/java/github/weichware10/analyse/gui/general/FunctionChooserController.java
+++ b/src/main/java/github/weichware10/analyse/gui/general/FunctionChooserController.java
@@ -21,7 +21,6 @@ public class FunctionChooserController extends AbsSceneController {
 
     @FXML
     void logOut() {
-        Logger.info("analyzer:content Resetting");
         Analyzer.reset();
         Logger.info("functionchooser:content Logging out");
         Login.logOut();

